activemq-broker/src/main/java/org/apache/activemq/security/jwt/JwtAuthenticationBroker.java

```diff
@@ -0,0 +1,179 @@
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Handle authentication an user based on a JWT token
+ */
+public class JwtAuthenticationBroker extends AbstractAuthenticationBroker {
+
+    private final String jwtIssuer;
+    private final Claims jwtGroupsClaim;
+    private final String jwtValidatingPublicKey;
+
+    public JwtAuthenticationBroker(
+            final Broker next,
+            final String jwtIssuer,
+            final Claims jwtGroupsClaim,
+            final String jwtValidatingPublicKey) {
+        super(next);
+        this.jwtIssuer = jwtIssuer;
+        this.jwtGroupsClaim = jwtGroupsClaim;
+        this.jwtValidatingPublicKey = jwtValidatingPublicKey;
+    }
+
+    @Override
+    public void addConnection(final ConnectionContext context, final ConnectionInfo info) throws Exception {
+        SecurityContext securityContext = context.getSecurityContext();
+        if (securityContext == null) {
+            securityContext = authenticate(info.getUserName(), info.getPassword(), null);
+            context.setSecurityContext(securityContext);
+            securityContexts.add(securityContext);
+        }
+
+        try {
+            super.addConnection(context, info);
+        } catch (Exception e) {
+            securityContexts.remove(securityContext);
+            context.setSecurityContext(null);
+            throw e;
+        }
+    }
+
+    @Override
+    public SecurityContext authenticate(final String username, final String password, final X509Certificate[] certificates) throws SecurityException {
+        SecurityContext securityContext = null;
+        if (!username.isEmpty()) {
+
+            // parse the JWT token and check signature, validity, nbf
+            try {
+                final JwtConsumerBuilder builder = new JwtConsumerBuilder()
+                        .setRelaxVerificationKeyValidation()
+                        .setRequireSubject()
+                        .setSkipDefaultAudienceValidation()
+                        .setRequireExpirationTime()
+                        .setExpectedIssuer(jwtIssuer)
+                        .setAllowedClockSkewInSeconds(5)
+                        .setVerificationKey(parsePCKS8(jwtValidatingPublicKey))
+                        .setJwsAlgorithmConstraints(
+                                new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.WHITELIST,
+                                        AlgorithmIdentifiers.RSA_USING_SHA256,
+                                        AlgorithmIdentifiers.RSA_USING_SHA384,
+                                        AlgorithmIdentifiers.RSA_USING_SHA512)
+                        );
+
+                final JwtConsumer jwtConsumer = builder.build();
+                final JwtContext jwtContext = jwtConsumer.process(username);
```

Comment:
On [the Jira](https://issues.apache.org/jira/browse/AMQ-9244?focusedCommentId=17723012&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17723012) it was indicated that the _password_ field should contain the token. If `username` is used instead will this fit the reported use-case?